src/main/java/com/blamejared/compat/tcomplement/highoven/HighOven.java

```diff
@@ -0,0 +1,305 @@
+			} else if (REMOVED_HEAT_RECIPES.containsKey(output)) {
+				Set<FluidStack> inputs = REMOVED_HEAT_RECIPES.get(output);
+				if (inputs != null)
+					inputs.add(InputHelper.toFluid(input));
+			} else {
+				Set<FluidStack> inputs = new HashSet<>();
+				inputs.add(InputHelper.toFluid(input));
+				REMOVED_HEAT_RECIPES.put(output, inputs);
+			}
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Removing %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		public String getRecipeInfo() {
+			return LogHelper.getStackDescription(output)
+					+ ((input == null) ? "" : (" from " + LogHelper.getStackDescription(input)));
+		}
+
+	}
+
+	@ZenMethod
+	public static void removeMixRecipe(ILiquidStack output, @Optional ILiquidStack input) {
+		init();
+		CraftTweakerAPI.apply(new RemoveMix(output, input));
+	}
+
+	@ZenMethod
+	public static MixRecipeBuilder newMixRecipe(ILiquidStack output, ILiquidStack input, int temp) {
+		init();
+		return new MixRecipeBuilder(output, input, temp);
+	}
+	
+	@ZenMethod
+	public static MixRecipeManager manageMixRecipe(ILiquidStack output, ILiquidStack input) {
+		init();
+		return new MixRecipeManager(output, input);
+	}
+
+	private static class RemoveMix extends BaseAction {
+		private ILiquidStack output;
+		private ILiquidStack input;
+
+		public RemoveMix(ILiquidStack output, ILiquidStack input) {
+			super("HighOven.MixRecipe");
+			this.output = output;
+			this.input = input;
+		}
+
+		@Override
+		public void apply() {
+			FluidStack output = InputHelper.toFluid(this.output);
+			if (this.input == null) {
+				REMOVED_MIX_RECIPES.put(output, null);
+			} else if (REMOVED_MIX_RECIPES.containsKey(output)) {
+				Set<FluidStack> inputs = REMOVED_MIX_RECIPES.get(output);
+				if (inputs != null)
+					inputs.add(InputHelper.toFluid(input));
+			} else {
+				Set<FluidStack> inputs = new HashSet<>();
+				inputs.add(InputHelper.toFluid(input));
+				REMOVED_MIX_RECIPES.put(output, inputs);
+			}
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Removing %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		protected String getRecipeInfo() {
+			return LogHelper.getStackDescription(output)
+					+ ((input == null) ? "" : (" from " + LogHelper.getStackDescription(input)));
+		}
+	}
+
+	@SubscribeEvent
+	public void onTinkerRegister(TCompRegisterEvent.HighOvenFuelRegisterEvent event) {
+		if (event.getRecipe() instanceof HighOvenFuelTweaker) {
+			return;
+		}
+		for (IItemStack entry : REMOVED_FUELS) {
+			if (event.getRecipe().matches(InputHelper.toStack(entry))) {
+				event.setCanceled(true);
+			}
+		}
+	}
+
+	@SubscribeEvent
+	public void onTinkerRegister(TCompRegisterEvent.HighOvenHeatRegisterEvent event) {
+		if (event.getRecipe() instanceof HeatRecipeTweaker) {
+			return;
+		} else {
+			for (Map.Entry<FluidStack, Set<FluidStack>> entry : REMOVED_HEAT_RECIPES.entrySet()) {
```

Comment:
I assume the issue with FluidStack to FluidStack was duplicate fluids deleted some combinations. I would suggest simplifying this by using `List<Pair<FluidStack,FluidStack>>` as that gets rid of the duplicate key problem. Just make sure you use the right pair so it exists on both client and server, the Inspirations cauldron class should have an example.

Reply:
Not really. This is to handle multiple inputs yielding the same output, if such behaviour is added by addons or other mods.
In case the user wants to delete all recipes for the given output, no matter the input, the input set is set to `null` and the method won't bother iterating.
If only some specific input should be disabled, the set is defined, inputs are stored and compared against during event handling.

Reply:
Yes, but there really is no need to do it as a set, it does not matter if I cancel on input A, output then 5 entries later cancel on input B, output; just check if any pair in the list matches and return after the first match. Plus the odds of a user removing more than 1 input without wanting wildcard is pretty low, since the pair method already handles it no reason to make it inefficient in most cases for negligible benefit in one. On a related note, pretty sure you will get a NPE in the CT method if you first use null input then a specific one, and even if not I'm not even certain fluidstacks support being a hashmap key

Additionally, the map misses the case of null output and defined input as maps don't allow null as a key last I checked. Just make it a list of pairs (or simpleentry) and it handles the problem you are trying to solve with far simplier code.

Reply:
Oh, you're right ! I changed that in the latest commit